contents/kotlin/kotlin.md

```diff
@@ -0,0 +1,87 @@
+# Introduction to Kotlin
+
+Author: [Shradheya Thakre](https://github.com/tshradheya)
+
+**Table of Contents**
+
+* [Overview](#overview)
+* [Why Kotlin?](#why-kotlin)
+    * [Concise](#concise)
+    * [Safe](#safe)
+    * [Interoperable](#interoperable)
+    * [Tool Friendly](#tool-friendly)
+* [Kotlin for Android Apps](#kotlin-for-android-apps)
+    * [Advantages of Shifting to Kotlin](#advantage-kotlin)
+    * [Drawbacks of Shifting to Kotlin](#drawbacks-kotlin)
+
+
+
+
+
+
+# Overview
+
+Kotlin is a general purpose, open source, statically typed “pragmatic” programming language for the JVM and Android that combines object-oriented and functional programming features.
+
+# Why Kotlin?
+
+### Concise
+
+Highly reduces the amount of boilerplate code.
+- A POJO with getters, setters, equals(), hashCode(), toString() and copy() can be created in a single line
+``` kotlin
+data class Customer(val name: String, val email: String, val company: String)
+```
+- Assigning values based on ranges is much simpler
+``` kotlin
+val quartile: Int
+quartile = when (playPercentage) {
+    in 0..24 -> 0
+    in 25..49 -> 1
+    ...
+}
+```
+
+
+### Safe
+Kotlin protects you from mistakenly operating on nullable types
+- Get compilation error when you mistakenly try operating on nullable types
+```
+val name: String? = null    // Nullable type
+println(name.length())      // Compilation error
+```
+
+### Interoperable
+
+- Target either the JVM or JavaScript. Write code in Kotlin and decide where you want to deploy
+```
+import kotlin.browser.window
+fun onLoad() {
+    window.document.body!!.innerHTML += "<br/>Hello, Kotlin!"
+}
+```
+- You can literally continue work on your old Java projects using Kotlin. All your favorite Java frameworks are still available, and whatever framework you’ll write in Kotlin can be easily adopted by other Java developers
+
+### Tool Friendly
+
+Hyperbolically, a programming language is only as good as its tool support. This is why advantage with using Kotlin is that IntelliJ provides built-in language support. It also contains the aforementioned Java-to-Kotlin converter and code generators for Java and JavaScript from Kotlin code
```

Comment:
I think paraphrasing some of these can make it clearer. What do you think?

tool support -> is only as good as what its tools can provide.
why advantage with -> why the advantage of using Kotlin is the built-in language support from IntelliJ

Kotlin code -> Kotlin code. (add a fullstop)

Reply:
Some comments

1. Is "Hyperbolically" suppose to be "Hypothetically" instead?
2. Better to also state Kotlin support for other commonly used IDEs like Eclipse, so that the scope of "tool-friendliness" can be more general. That way Kotlin would seem even more accessible